lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -2483,7 +2483,9 @@ Value *PatchInOutImportExport::patchFsBuiltInInputImport(Type *inputTy, unsigned
                                             {ancillary, builder.getInt32(8), builder.getInt32(4)});
 
     Value *sampleMaskIn = sampleCoverage;
-    if (m_pipelineState->getRasterizerState().perSampleShading || builtInUsage.runAtSampleRate) {
+    // It should keep pixelShaderSamples be valid value, otherwise it will hang in while loop.
```

Comment:
This comment is not a good justification for why this change ought to be correct.

What is the justification for why this is correct? When can it happen that pixelShaderSamples == 0 but runAtSampleRate is true?

Reply:
Thanks for the comments. I have some investigation before:

From some comments in code:
```
runAtSampleRate: Whether fragment shader run at sample rate
pixelShaderSamples:  Controls the pixel shader execution rate. Must be less than or equal to coverageSamples. Valid values are 1, 2, 4, and 8.
```
The pixelShaderSamples will be set from driver when BuildMultisampleState, seems the value for this will be set from API.
But the runAtSampleRate will be set to 1 when read builtInvariable(BuiltInSampleId/BuiltInSamplePosition/..) or have InterpLocSample infos, which is set when compile shader.

So in next product, broadcast samplemask will only be used when pixelShaderSamples is set.

I will expand the comments.